chatrooms.lua

```diff
@@ -51,35 +65,72 @@ function SaveToFile( sChatMessage, sRoom )
 end
 
 function SendToRoom( tSelfUser, sRoom, sIncoming )
-	for iIterate = 0, tChatRooms[sRoom].iMaxProfile do
-		local tUsers = Core.GetOnlineUsers( iIterate )
-		if tUsers then
-			for iIndex, tRecipient in ipairs( tUsers ) do
-				if tRecipient.sNick:lower() ~= tSelfUser.sNick:lower() then
-					Core.SendToUser( tRecipient, "$To: "..tRecipient.sNick.." From: "..sRoom.." $"..sIncoming.."|" )
-				end
+	local tCurrentHistory = tChatRooms[sRoom].tChatHistory
+	table.insert( tCurrentHistory, os.date(tConfig.sTimeFormat)..sIncoming )
+	if tCurrentHistory[ tConfig.iMaxHistory + 1 ] do
+		table.remove( tCurrentHistory, 1 )
+	end
+	local tUsers = tChatRooms[sRoom].tUsers
+	if tUsers then
```

Comment:
`tUsers` will always be a non-`nil` value, since it is hard-coded to be a table. This particular `if` block can be removed. If there are no elements inside the table, `ipairs` will return `nil` and the for loop will not execute.